packages/elements/src/tree/managers/tree-manager.ts

```diff
@@ -406,6 +407,7 @@ export class TreeManager<T extends TreeDataItem> {
   }
   private _uncheckItem(item: T, manageRelationships = this.manageRelationships): boolean {
     if (this.canUncheckItem(item)) {
+      delete item.selectedAt;
```

Comment:
I think we don't need to delete selctedAt and leave it as is when the item is unchecked. 
If an item is unchecked and then rechecked, the selectedAt timestamp would reflect the most recent selection time. Moreover delete method aslo has a significant impact on performance.

Reply:
Removed